integ/workflow/parallel/routers.go

```diff
@@ -71,7 +72,11 @@ func (h *handler) ApiV1WorkflowStateStart(c *gin.Context, t *testing.T) {
 	log.Println("received state start request, ", req)
 
 	if req.GetWorkflowType() == WorkflowType {
-		h.invokeHistory[req.GetWorkflowStateId()+"_start"]++
+		if value, ok := h.invokeHistory.Load(req.GetWorkflowStateId() + "_start"); ok {
+			h.invokeHistory.Store(req.GetWorkflowStateId()+"_start", value.(int64)+1)
+		} else {
+			h.invokeHistory.Store(req.GetWorkflowStateId()+"_start", int64(1))
+		}
```

Comment:
I couldn't reproduce the issue described, but I can see that this issue could very well happen due to the nature of the Workflow in this test.

Looking for opinions on if this is something that we need to polish a bit more, and make all our other tests also use `sync.Map` for storing the current `invokeHistory` in the integration tests

Reply:
As discussed, I have applied the same logic now for all the tests. i.e. all tests now use `sync.Map` which can be written to concurrently